packages/client/hmi-client/src/components/workflow/scenario-templates/calibrate-ensemble/tera-calibrate-ensemble-template.vue

```diff
@@ -0,0 +1,245 @@
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import { AssetType, Dataset, ModelConfiguration } from '@/types/Types';
+import { useProjects } from '@/composables/project';
+import { getModel, getModelConfigurationsForModel } from '@/services/model';
+import { isEmpty } from 'lodash';
+import Dropdown from 'primevue/dropdown';
+import Button from 'primevue/button';
+import TabView from 'primevue/tabview';
+import TabPanel from 'primevue/tabpanel';
+import TeraInputText from '@/components/widgets/tera-input-text.vue';
+import { getFileName } from '@/services/calibrate-workflow';
+import { getDataset } from '@/services/dataset';
+import CalibrateEnsembleImg from '@/assets/svg/template-images/calibrate-ensemble-model-thumbnail.svg';
+import { ScenarioHeader } from '../base-scenario';
+import { CalibrateEnsembleScenario } from './calibrate-ensemble-scenario';
+import TeraScenarioTemplate from '../tera-scenario-template.vue';
+
+const header: ScenarioHeader = Object.freeze({
+	title: 'Calibrate an ensemble model template',
+	question: 'Create a more accurate model by combining multiple models in an ensemble.',
+	description: 'Simulate and calibrate each model individually, then calibrate the ensemble.',
+	examples: ['How can I leverage the strengths of each model to make the most accurate model possible?']
+});
+
+const datasets = computed(() => useProjects().getActiveProjectAssets(AssetType.Dataset));
+const models = computed(() => useProjects().getActiveProjectAssets(AssetType.Model));
+const isFetchingModelInformation = ref(false);
+
+const modelAssets = ref<Map<string, { modelConfigurations: ModelConfiguration[]; modelStates: any }>>(new Map());
+
+const selectedModelConfigurations = computed<ModelConfiguration[]>(() =>
+	// get the model configuration ids from the tab specs and return the already fetched model configurations from the model assets
+	props.scenario.tabSpecs
+		.map((tab) =>
+			modelAssets.value
+				.get(tab.modelSpec.id)
+				?.modelConfigurations.find((config) => config.id === tab.modelConfigSpec.id)
+		)
+		.filter((config) => config !== undefined)
+);
+
+const dataset = ref<Dataset | null>();
+
+const tableHeaders = computed(() => {
+	const headers = ['Ensemble model'];
+
+	headers.push(dataset.value ? getFileName(dataset.value) : '');
+
+	selectedModelConfigurations.value.forEach((config) => {
+		headers.push(config.name ?? '');
+	});
+	return headers;
+});
+
+const datasetColumnNames = computed(() => dataset.value?.columns?.map((col) => col.name) ?? ([] as string[]));
+
+const activeTab = ref(0);
+const props = defineProps<{
+	scenario: CalibrateEnsembleScenario;
+}>();
+
+const allModelsIds = computed(() => props.scenario.tabSpecs.map((tab) => tab.modelSpec.id));
+
+const emit = defineEmits(['save-workflow']);
+
+const onRemoveTab = (index: number) => {
+	props.scenario.removeTabSpec(index);
+	if (activeTab.value > 0 && index === activeTab.value) {
+		activeTab.value--;
+	}
+};
```

Comment:
This isn't quite correct, what is the desired effect here?

If I have 
- Tab1
- Tab2
- Tab3
- Tab4

I am at Tab4 and I remove Tab2, the UI seems to get a bit messed up
<img width="383" alt="image" src="https://github.com/user-attachments/assets/a173d751-3041-4461-85c1-331fadc8050c" />